cpp/src/compression/benchmark/ParquetBenchmark.cpp

```diff
@@ -68,18 +71,99 @@ class ParquetBenchamarkReader {
   int64_t total_byte_size_;
 };
 
-static void BM_ParquetRead(::benchmark::State& state) {
-  ParquetBenchamarkReader reader;
-  state.counters["ratio"] = static_cast<double>(reader.getTotalByteSize()) /
-                            static_cast<double>(reader.getTotalCompressedSize());
+int SetCurrentThreadAffinity(int cpu) {
+  cpu_set_t cpuset;
+  CPU_ZERO(&cpuset);
+  CPU_SET(cpu, &cpuset);
+  return pthread_setaffinity_np(pthread_self(), sizeof(cpu_set_t), &cpuset);
+}
+
+void BM_ParquetRead(benchmark::State& state, const std::string& file_name) {
+  SetCurrentThreadAffinity(state.thread_index() + 1);
 
+  double total_seconds = 0;
   for (auto _ : state) {
     state.PauseTiming();
+    ParquetReader reader(file_name);
     state.ResumeTiming();
+
+    auto start = std::chrono::steady_clock::now();
     reader.read();
+    auto end = std::chrono::steady_clock::now();
+    total_seconds +=
+        std::chrono::duration_cast<std::chrono::duration<double>>(end - start).count();
   }
 
-  state.SetBytesProcessed(state.iterations() * reader.getTotalCompressedSize());
+  ParquetReader reader(file_name);
+  reader.readMetadata();
+
+  state.SetBytesProcessed(state.iterations() * reader.getTotalByteSize());
+
+  state.counters["bw"] =
+      benchmark::Counter(state.iterations() * reader.getTotalByteSize() / total_seconds,
+                         benchmark::Counter::kAvgThreads,
+                         benchmark::Counter::OneK::kIs1024);
+
+  double ratio = static_cast<double>(reader.getTotalByteSize()) /
+                 static_cast<double>(reader.getTotalCompressedSize());
+  state.counters["ratio"] = benchmark::Counter(
+      ratio, benchmark::Counter::kAvgThreads, benchmark::Counter::OneK::kIs1000);
+}
+
+void print_usage() {
+  std::cout << R"(Usage:
+    '--iter'   or '-t' to specify the number of iterations to run the benchmark
+    '--thread' or '-t' to specify the number of threads to run the benchmark
+    '--file'   or '-f' to specify the name of parquet file to run the benchmark
+    '--codec'  or '-c' to specify the backend codec to run the benchmark
+    '--help'   or '-h' to print the help)"
+            << std::endl;
+  return;
 }
 
-BENCHMARK(BM_ParquetRead);
+}  // namespace
+
+int main(int argc, char** argv) {
+  int iterations = 1;
+  int threads = 1;
+  std::string file = "lineitem.igzip";
+  std::string codec = "igzip";
+
+  for (auto i = 0; i < argc; i++) {
```

Comment:
small tip: you can use gflags as command line option.

Reply:
I think no need to introduce gflags  as dependency